### ISSUE TITLE
False error about being not multithreading safe on OS X

### ISSUE THREAD

**Issue by cb0** [OfflineIMAP/offlineimap3#141]
#### General informations

- system/distribution (with version): 
  MacOS Monterey 12.5.1
  Darwin 21.6.0 Darwin Kernel Version 21.6.0: Wed Aug 10 14:25:27 PDT 2022; root:xnu-8020.141.5~2/RELEASE_X86_64 x86_64

- offlineimap version (`offlineimap -V`): 
  offlineimap v8.0.0, imaplib2 v3.06, Python v3.11.0, OpenSSL 1.1.1s  1 Nov 2022
- Python version: 
  Python 3.10.8

#### Error

Running the cli offlineimap command results in this error

```
ERROR: While attempting to sync account 'account@....'
  Your sqlite is not multithreading safe.

Traceback:
  File "/usr/local/Cellar/offlineimap/8.0.0_1/libexec/lib/python3.11/site-packages/offlineimap/accounts.py", line 298, in syncrunner
    self.__sync()
  File "/usr/local/Cellar/offlineimap/8.0.0_1/libexec/lib/python3.11/site-packages/offlineimap/accounts.py", line 377, in __sync
    remoterepos.sync_folder_structure(localrepos, statusrepos)
  File "/usr/local/Cellar/offlineimap/8.0.0_1/libexec/lib/python3.11/site-packages/offlineimap/repository/Base.py", line 292, in sync_folder_structure
    status_repo.makefolder(local_name.replace(
  File "/usr/local/Cellar/offlineimap/8.0.0_1/libexec/lib/python3.11/site-packages/offlineimap/repository/LocalStatus.py", line 123, in makefolder
    folder.openfiles()
  File "/usr/local/Cellar/offlineimap/8.0.0_1/libexec/lib/python3.11/site-packages/offlineimap/folder/LocalStatusSQLite.py", line 108, in openfiles
    assert sqlite.threadsafety == 2, 'Your sqlite is not multithreading safe.'

```

I checked sqlite version and it is thread safe:

```
echo 'PRAGMA compile_options;' | sqlite3  | grep THREADSAFE                                                            
THREADSAFE=2

```

I'm not sure, but the following change solved my issue. 

Change (offlineimap/folder/LocalStatusSQLite.py)

    def openfiles(self):
        # Make sure sqlite is in multithreading SERIALIZE mode.
        assert sqlite.threadsafety == 1, 'Your sqlite is not multithreading safe.'

into 

    def openfiles(self):
        # Make sure sqlite is in multithreading SERIALIZE mode.
        assert sqlite.threadsafety >= 1, 'Your sqlite is not multithreading safe.'

According to the docs, 3 means 

"Serialized. In serialized mode, SQLite can be safely used by multiple threads with no restriction."

I could create a PR if you find this solution acceptable.

**Comment by MichaelAnckaert** [comment]
Chiming in here:

% offlineimap -V
offlineimap v8.0.0, imaplib2 v3.06, Python v3.11.0, OpenSSL 1.1.1s  1 Nov 2022

% uname -a
Darwin Michaels-MacBook-Pro-2.local 21.3.0 Darwin Kernel Version 21.3.0: Wed Jan  5 21:37:58 PST 2022; root:xnu-8019.80.24~20/RELEASE_ARM64_T6000 arm64

I applied the change above and can confirm that `offlineimap` works as expected.

**Comment by junnu-github** [comment]
Isn't this already solved by the commit https://github.com/OfflineIMAP/offlineimap3/commit/7cd32cf834b34a3d4675b29bebcd32dc1e5ef128 ?

**Comment by cb0** [comment]
> Isn't this already solved by the commit [7cd32cf](https://github.com/OfflineIMAP/offlineimap3/commit/7cd32cf834b34a3d4675b29bebcd32dc1e5ef128) ?

It seem like it is. I wasn't aware of that one, seem my OS version is not yet up to date or this isn't merged yet. But anyway, this seems to solve the issue too. I'll therefore close this one.
